src/xdg-autostart-generator/xdg-autostart-condition.c

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "main-func.h"
+#include "strv.h"
+
+static int run(int argc, char *argv[]) {
+        _cleanup_strv_free_ char **only_show_in = NULL;
+        _cleanup_strv_free_ char **not_show_in = NULL;
+        _cleanup_strv_free_ char **desktops = NULL;
+        char *xdg_current_desktop;
```

Comment:
i'd prefix this with "const". doesn't matter, but is kinda nice given that we typically don't want to make changes to env vars in place